### ISSUE TITLE
OneOf not processed properly in C# generator

### ISSUE THREAD

**Issue by mol-pensiondk** [OpenAPITools/openapi-generator#16554]
In https://github.com/OpenAPITools/openapi-generator/blob/78f3b19b58df699ef883b89a7a44531407377719/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java#L530C80-L530C85 the anyOf should be oneOf in the subcondition (model.anyOf != null && model.oneOf.size() > 0), i.e. it should read (model.**oneOf** != null && model.oneOf.size() > 0).

**Comment by wing328** [comment]
Thanks for reporting the issue.

Can you please file a PR with the suggested fix? Let me know if you need help with the PR.

**Comment by mol-pensiondk** [comment]
I don't have a java development environment, so would not be able to check that my change builds and does not break any tests. I could probably make a "blind" PR regardless, if that would make it easier to handle the issue? For #16555 and this one both?

**Comment by wing328** [comment]
yes please file one. we will test it to ensure the change is good.

**Comment by mol-pensiondk** [comment]
Ok, tried my hand at creating a pull request directly in the github UI.
https://github.com/OpenAPITools/openapi-generator/pull/16577
Hope I did not break too many rules...
